etcd/watchmanager/types.go

```diff
@@ -35,7 +35,7 @@ type WatchManager interface {
 }
 
 // UpdateFn is called when an event on the watch channel happens
-type UpdateFn func(key string) error
+type UpdateFn func(key string, events []*clientv3.Event) error
```

Comment:
Sure, I can split them up. I didn't want to do that initially because I wasn't sure to do with the case where the event slice was not nil but just empty. I guess we can just skip calling the updateFn and emit a warning or something in that case.